File: auth-application/src/main/java/com/example/auth/application/service/ListMySessionsService.java
package com.example.auth.application.service;

import com.example.auth.application.port.in.ListMySessionsUseCase;
import com.example.auth.application.port.out.RefreshTokenRepository;
import com.example.auth.domain.common.TenantId;
import com.example.auth.domain.common.UserId;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
public class ListMySessionsService implements ListMySessionsUseCase {

    private final RefreshTokenRepository refreshTokenRepository;

    @Override
    @Transactional(readOnly = true)
    public List<SessionView> list(TenantId tenantId, UserId userId) {
        return refreshTokenRepository.findActiveByUser(tenantId, userId).stream()
                .map(t -> new SessionView(
                        t.id(),
                        t.deviceLabel(),
                        t.ipAddress(),
                        t.issuedAt(),
                        t.lastUsedAt(),
                        t.expiresAt()))
                .toList();
    }
}

File: auth-application/src/main/kotlin/com/example/auth/application/service/ListMySessionsService.kt
package com.example.auth.application.service

import com.example.auth.application.port.`in`.ListMySessionsUseCase
import com.example.auth.application.port.`in`.ListMySessionsUseCase.SessionView
import com.example.auth.application.port.out.RefreshTokenRepository
import com.example.auth.domain.common.TenantId
import com.example.auth.domain.common.UserId
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class ListMySessionsService(
    private val refreshTokenRepository: RefreshTokenRepository,
) : ListMySessionsUseCase {

    @Transactional(readOnly = true)
    override fun list(tenantId: TenantId, userId: UserId): List<SessionView> =
        refreshTokenRepository.findActiveByUser(tenantId, userId).map { t ->
            SessionView(
                t.id,
                t.deviceLabel,
                t.ipAddress,
                t.issuedAt,
                t.lastUsedAt,
                t.expiresAt,
            )
        }
}
